### DISCUSSION TITLE
列表长文字展示

### DISCUSSION THREAD

**Discussion by slowlyo** [slowlyo/owl-admin#65] [组件封装]
```php
public function listLongText($name, $label, int $limit = 20)
{
    return amis()->TableColumn()
        ->name($name)
        ->label($label)
        ->type('tpl')
        ->tpl('${' . $name . '|truncate:' . $limit . '}')
        ->popOver(amis()->SchemaPopOver()->body('${' . $name . '}')->trigger('hover'));
}
```

**Comment by jasonup2020** [comment]
/**
     * 列表长文字展示自动隐藏 #65  Components::make()->listLongText('firstname', '产品型号','${firstname}', 10)->remark('long...')->width(20),
     * @param type $name 标签名EN
     * @param type $label  标签名CN
     * @param type $row 参数值默认Name同名 格式如 '${name}'
     * @param int $limit 默认>=20字数
     * @return type
     */
    public function listLongText($name, $label, $row = "", int $limit = 20) {
        // 1️⃣ 获取当前字段值（如果存在）
        $value = $row ?? '';
        // 2️⃣ 安全计算长度（支持中文）
        $length = iconv_strlen($value, 'UTF-8');
        // 3️⃣ 判断是否超过限制
        if ($length < $limit) {
            // 超出长度，截断显示 + hover 弹出完整文本
            return amis()->TableColumn()
                            ->name($name)
                            ->label($label)
                            ->type('tpl')
                            ->tpl('${' . $name . '|truncate:' . $limit . '}')
                            ->popOver(
                                    amis()->SchemaPopOver()
                                    ->body('${' . $name . '}')
                                    ->trigger('hover')
//                                    ->showIcon(false) #是否显示放大镜
            );
        } else {
            // 未超出长度，正常显示
            return amis()->TableColumn()
                            ->name($name)
                            ->label($label)
                            ->type('tpl')
                            ->tpl('${' . $name . '}');
        }
    }
